libvast/src/system/query_supervisor.cpp

```diff
@@ -87,8 +87,8 @@ query_supervisor_actor::behavior_type query_supervisor(
       for (const auto& [id, partition] : qm) {
         auto partition_trace_id = id.as_u64().first;
         ++*open_requests;
-        // TODO: Add a proper configurable timeout.
-        self->request(partition, caf::infinite, query)
+        self
+          ->request(partition, defaults::system::partition_query_timeout, query)
```

Comment:
NOTE: This causes a shutdown hang in the VAST server process for the timeout period. Logging reveals that both query supervisors and partitions shut down properly, so I assume that this is in the CAF actor clock itself.

I'll cherry-pick the remaining commits into a separate PR.